core/unix/signal.c

```diff
@@ -4838,12 +4853,24 @@ master_signal_handler_C(byte *xsp)
                 dcontext = NULL;
             }
         }
-        if (dcontext == NULL) {
-            /* Check for a temporarily-native thread we're synch-ing with. */
-            tr = thread_lookup(get_sys_thread_id());
-            if (tr != NULL)
-                dcontext = tr->dcontext;
-        }
+    }
+    if (dcontext == NULL &&
+        /* Check for a temporarily-native thread we're synch-ing with. */
+        (sig == SUSPEND_SIGNAL
+#ifdef X86
```

Comment:
I don't remember what ARM does (when will dcontext be NULL there): maybe worth a XXX comment.

Reply:
Do you want a new issue or even leave i#2921 open for this, or just XXX?

Reply:
No strong opinion, whatever you want to do.

Reply:
Done.